include/secp256k1_rangeproof.h

```diff
@@ -287,6 +287,33 @@ SECP256K1_API SECP256K1_WARN_UNUSED_RESULT int secp256k1_rangeproof_info(
   size_t plen
 ) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3) SECP256K1_ARG_NONNULL(4) SECP256K1_ARG_NONNULL(5);
 
+/** Returns the maximum size of a rangeproof with the given parameters
+ *
+ * An actual rangeproof may be smaller, for example if the actual value
+ * is less than both the provided `max_value` and 2^`min_bits`, or if
+ * the `exp` parameter to `secp256k1_rangeproof_sign` is set such that
+ * the proven range is compressed. In particular it will always
+ * overestimate the size of single-value proofs. Also, if `min_value`
+ * is set to 0 in the proof, the result will usually, but not always,
+ * be 8 bytes smaller than if a nonzero value had been passed.
+ *
+ * The goal of this function is to provide a useful upper bound for
+ * memory allocation or fee estimation purposes, without requiring
+ * too many parameters be fixed in advance.
+ *
+ * To obtain the size of largest possible proof, set `min_value` to 1 and
+ * `max_value` to UINT64_MAX.
```

Comment:
ah no, this function doesn't even have a `min_value` arg :P

Reply:
Oops, it originally did, which would save a few bytes when `min_value` was 0 ... but then I realized that sometimes `rangeproof_sign` would override the user's choice of 0 if `exp` was also set, so I gave up on it.

BTW @jonasnick I wonder if this addresses your "sometimes wrong values when `min_value` is 0" observation?

Reply:
@real-or-random I will remove the reference to `min_value` but I think the existing text "size of the largest possible proof" is correct and what we want to say.

An "upper bound" is too weak, and it seems weird to mention `min_value` and `max_value` without mentioning `ct_bits` or `exp` or any of the other parameters to `rangeproof_sign` which could affect the size.

Reply:
@apoelstra I wrote `max_value` but I meant `min_bits` (it's right in th ebranch_

Reply:
@jonasnick ah, yep, I see

Reply:
To be honest, I think my problem here is that I don't understand the purpose of `max_value` at all, or maybe that I don't understand the API of `rangeproof_sign`

So I guess here `max_value` is supposed limit of the range covered by the proof? But how do I specify the upper limit of the range in `rangeproof_sign`? Only via  `min_bits`? This is not clear to me from the docs of `rangeproof_sign`

Reply:
@real-or-random it's implied by the union of `value` and `min_bits`, both of which are provided to this function.

Reply:
Ok, how it is implied? I don't see it. Shouldn't the upper limit of the range be *independent* of `value`? I mean that's the entire point of a rangeproof?

Sorry, I must be missing something,  I guess I'm having a mental block here.... ^^

Reply:
It's not independent because gmax's code will increase the range if the value would otherwise be outside of it.

Reply:
Oh I see. That was totally unclear to me from the docs. I think we should clarify this in the docs of `rangeproof_sign`, ideally by describing `max_value` as mathematical function of `value` and `min_bits`...   

Also this is unclear to the reader in the docs.
```min_bits: Number of bits of the value to keep private. (0 = auto/minimal, - 64).```
 * which bits? most significant or least significant?
 * what's "auto/minimal" supposed to do?

But this can be done in another PR.

Reply:
>That was totally unclear to me from the docs. 

Lol. We could add something to `rangeproof_sign` to the effect of "other than `value` and `blind` basically every value may be overridden by the signing logic, in ad-hoc ways, if the proof otherwise would be impossible to construct.

>ideally by describing max_value as mathematical function of value and min_bits

I mean, this function is a somewhat arbitrary choice of function here. It won't match the behavior of `rangeproof_sign` because `rangeproof_sign` also takes its other arguments into account.

> which bits? most significant or least significant?

All of them. If `value` is too large the code just increases `min_bits`, unless `exp` is -1 (IIRC) in which case it will create a single-value proof.

In general I think it's a fool's errand to try to document the existing rangeproof signer logic (it may be feasible to just rip it out, respect the user's params, and refuse to sign if they're incompatible, but it's quite hard to communicate what needs to be adjusted, which I think is why gmax made the choices he did). What I need right now is a way for Elements to swag the size of a rangeproof, for fee estimation, given that it knows the user's choice of `min_bits` and that any output value will be <= `MAX_MONEY`.

Reply:
> > ideally by describing max_value as mathematical function of value and min_bits
> 
> I mean, this function is a somewhat arbitrary choice of function here. It won't match the behavior of `rangeproof_sign` because `rangeproof_sign` also takes its other arguments into account.

Ah I see. Sorry, I misunderstood. I assumed the max value in `rangeproof_sign` only depends on these two arguments because you said "it's implied by the union of `value` and `min_bits`".

Anyway, this PR here is fine. I think the docs of `rangeproof_sign` can be improved but if anything, that's a separate issue.